powershell/VstsTaskSdk/InputFunctions.ps1

```diff
@@ -158,60 +160,121 @@ function Get-EnvVariable {
         }
     }
 
-    if ($item -and $item.value) {
-        $value = $item.value
-        Write-Verbose "$($path): '$value'"
+    Get-FinalValue -Value $item.Value @PSBoundParameters
+}
+
+function Get-SecureInput {
+    [CmdletBinding(DefaultParameterSetName = 'Require')]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Name,
+        [Parameter(Mandatory = $true)]
+        [string]$Path,
+        [Parameter(ParameterSetName = 'Require')]
+        [switch]$Require,
+        [Parameter(ParameterSetName = 'Default')]
+        [object]$Default,
+        [switch]$AsBool,
+        [switch]$AsInt)
+
+    # Attempt to get the secure variable.
+    $value = $null
+    if ($psCredential = $script:secureInputs[$Path]) {
+        $value = $psCredential.GetNetworkCredential().Password
```

Comment:
Good question. I'll just confirmed it will work on my non-domain joined machine at home. FYI - I peeked into the .Net Core source code of GetNetworkCredential when I was adding this... It is surprisingly simple. It's basically just a class with three string properties (domain, username, pw). I had expected to find more magic :)